File: geeksforgeeks/Algorithms/Searching and Sorting/binary_search.java
import java.util.*;
public class Main {
	static int binarySearch(int arr[] , int l, int r , int x)
	{
		if(r>=l)
		{
			int mid = l + (r-l)/2;
			
			if(arr[mid]==x)
			{
				return mid;
			}
			
			if(arr[mid]>x)
			{
				return binarySearch(arr,l,mid-1,x);
			}
			else
			{
				return binarySearch(arr, mid+1, r, x);
			}
		}
		return -1;
	}
	
	static int binarySearchIterative(int arr[], int l, int r, int x) // Iterative Approach 
	{
	  while (l <= r)
	  {
	    int m = l + (r-l)/2;
	 
	    if (arr[m] == x) return m;  
	 
	    if (arr[m] < x) l = m + 1; 
	 
	    else r = m - 1; // 
	  }
	  return -1; // 
	}
	public static void main(String[] args) 
	{
		int arr[] = {2, 3, 4, 10, 40};
		int n = arr.length;
	    int x = 10;
  	    int result = binarySearch(arr, 0, n-1, x);
  	    
  	    if(result==-1)
  	    {
  	    	System.out.println("No index found");
  	    }
  	    else
  	    {
  	    	System.out.println("Value is present on index no: "+ result);
  	    }
	}  
}

File: geeksforgeeks/Top 10 Algorithms and Data Structures/Searching and Sorting/Binary Search.cpp
#include <iostream>
using namespace std;

int binarySearch(int a[],int start, int end, int num)
{
    if(end >= start)
    {
        int mid = start + (end - start)/2;
        
        if(a[mid] == num)
        {
            return mid;
        }
        else if(a[mid] < num)
        {
            return binarySearch(a, mid+1, end, num);
        }
        else
        {
            return binarySearch(a, start, mid-1, num);
        }
    }
    else
    {
        return -1;
    }
}

int main() {
	int a[] = {1,3,23,34,56,76,87,98,102};
	int num = 76;
	int l = sizeof(a)/sizeof(a[0]);
	
	if(binarySearch(a, 0, l-1, num) == -1)
	{
	    cout << num << " is not present in array.\n";
	}
	else
	{
	    cout << num << " is present at index: " << binarySearch(a, 0, l-1, num);
	}
	return 0;
}